app/assets/javascripts/angular/directives/frame.js.erb

```diff
@@ -66,13 +73,14 @@ angular.module("Prometheus.directives").directive('inlineFrame', ["$sce", "Varia
         });
 
         var height = frameHeight();
-        fields.height = height;
-        fields.width = (height / scope.aspectRatio) - FRAME_CONTAINER_PADDING;
-        fields.from = GraphiteTimeConverter.graphiteFrom(scope.frame.range, scope.frame.endTime);
-        fields.until = GraphiteTimeConverter.graphiteUntil(scope.frame.endTime);
-        fields.bgcolor = FRAME_BACKGROUND_COLOR;
-        parser.search = '?' + decodeURI($.param(fields)) + '&' + targets.join('&') + '&decache=' + scope.refreshCounter;
-        return parser.href;
+
+        return [
+          'height=' + height,
+          'width=' + ((height / scope.aspectRatio) - FRAME_CONTAINER_PADDING),
+          'from=' + GraphiteTimeConverter.graphiteFrom(scope.frame.range, scope.frame.endTime),
+          'until=' + GraphiteTimeConverter.graphiteUntil(scope.frame.endTime),
+          'bgcolor=' + FRAME_BACKGROUND_COLOR
+        ].concat(targets);
```

Comment:
Doesn't this change the behavior? There are more allowed fields in graphite than targets and the five defined above. We don't want to remove them. Can you explain the motivation of the change and how other fields like `ymax=500` will be preserved?